### PR TITLE
Basic Dropimplementation + Item not flying around like a bee and actually works.

### PR DESCRIPTION
**Pull request by steffengy**
Implemented basic drops of the entity, while they die.

### CODE REVIEW

File: src/main/java/org/spout/vanilla/controller/VanillaController.java [hunk 1]
```diff
@@ -75,12 +76,24 @@ public void onAttached() {
 		getParent().setCollision(new CollisionModel(area));
 		getParent().setData(VanillaControllerTypes.KEY, getType().getID());
 	}
+	
+	/**
+	 * Called like onDeath, but before the entity gets actually destroyed
+	 */
+	public void preDeath() {
```
Comment (windy1) [inline src/main/java/org/spout/vanilla/controller/VanillaController.java]: private void**

**Comment by Zidane** [comment]
Lets agree on a onDeath instead of preDeath (preDeath is kind of a false impression).

**Comment by steffengy** [comment]
As I remember there were two problems with using onDeath:
- The Entity didn't provide any Position
- was also called on Disconnect of Player

But If you're sure that using it, should be perfect Let me know.

**Comment by ZNickq** [comment]
The animation for dropped items IS done by the client, i implemented it myself, but when you initially destroy a block, the item jumps a little, check normal mc :P

**Comment by steffengy** [comment]
But your implementation forced the item to fly around in the air (like a crazy drunk bee) and as long as you don't implement it correctly you shouldn't break items completly ^^

**Comment by ZNickq** [comment]
No, i implemented the animation sending, the flight code was done by someone else to test entity movement, among others...

**Comment by steffengy** [comment]
Ah now i understand ^^

**Comment by Zidane** [comment]
Steffengy it should be fine. The issue with disconnecting players calling onDeath can be dealt with. Quite simply by seeing if the player no longer has an assigned network synchronizer should do so. The player's entity is removed before the Pre-Snapshot stage of the tick (if the player logged out) so it works out.

As for your initial point...can you elaborate?

**Comment by steffengy** [comment]
Will try to

**Comment by windy1** [comment]
Has this been resolved? If not, you need to update to the current HEAD ;)

**Comment by steffengy** [comment]
Already did once - this request was simply ignored then ^^

**Comment by stephenwilliams** [comment]
@steffengy it needs to be rebased to the current HEAD before @NinjaZidane pulls it

**Pull request merged**